src/sys/signal.rs

```diff
@@ -388,8 +388,12 @@ pub fn pthread_sigmask(how: SigFlags,
     Errno::result(res).map(drop)
 }
 
-pub fn kill(pid: libc::pid_t, signal: Signal) -> Result<()> {
-    let res = unsafe { libc::kill(pid, signal as libc::c_int) };
+pub fn kill(pid: libc::pid_t, signal: Option<Signal>) -> Result<()> {
```

Comment:
Trying to think about it differently: the change from `Signal` to `Option<Signal>` is breaking. The change from `Option<Signal>` to `T: Into<Option<Signal>>` is (I think?) not breaking thanks to `impl<T> From<T> for T`. (This depends on how you read "breaking" in signatures.) Our use of the `Into<Option<T>>` impl would be more akin to "progressive enhancement". A downstream crate requiring its use would have a 1.12+ minimum version, but we would not.